proto/spire/api/types/federationrelationship.proto

```diff
@@ -26,11 +26,13 @@ message HTTPSSPIFFEProfile {
     // Specifies the expected SPIFFE ID of the SPIFFE bundle endpoint server.
     string endpoint_spiffe_id = 1;
 
-    // The bundle of the trust domain that will be used when performing requests
-    // to the SPIFFE bundle endpoint server. It will be persisted for future requests.
-    // If there is an existing stored bundle, this bundle will overwrite it. If
-    // not specified, a previously stored bundle must exist.
-    spire.api.types.Bundle bundle = 2;
+    // The bundle used to bootstrap the connection to the SPIFFE bundle
+    // endpoint server. This field is required for self-serving endpoints (i.e.
```

Comment:
Is it required if a bundle is already set for the trust domain? E.g. Delete then re-create the relationship, or create the relationship then set later

Reply:
Currently, it is. Not sure it should be.

Reply:
The more I think about it, the less inclined I am to require the bundle in any situation. I'm ----><---- close to thinking maybe we just get rid of this field altogether but there is a convenience factor that maybe I'm not evaluating clearly.

Reply:
I've almost come full circle and want to get rid of the notion of self-serving v.s. non self-serving in our experience. I'm not sure what the spec would have to say about it, or if it is dangerous somehow. I'm leaning on two directions:

1. get rid of that field altogether. we can return a field in the response letting the caller know if the endpoint bundle is known or not when the relationship is made, or...
2. that field is named endpoint_bundle, (or endpoint_bootstrap_bundle), is optional, and if set, is used to create/replace the bundle for the endpoint trust domain. We would always return the endpoint bundle on a get/list if the bundle is known. (edited) 

We can always make the replace functionality opt-in or opt-out if we determine the correct default behavior but think it may not be universally wanted.

Reply:
We've discussed this in the SPIFFE slack. The updated change reflects the outcome.